File: LeetCode/121-best-time-to-buy-and-sell-stock/ex_121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minSoFar = INT_MAX;
        int maxProfit = 0;
        for (int n : prices) {
            maxProfit = max(maxProfit, n - minSoFar);
            minSoFar = min(minSoFar, n);
        }
        return maxProfit;
    }
};

File: LeetCode/121-best-time-to-buy-and-sell-stock/ex_121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min_so_far = sys.maxsize
        max_profit = 0
        for i in prices:
            max_profit = max(max_profit, i - min_so_far)
            min_so_far = min(min_so_far, i)
        return max_profit
        